docs/architecture/application/scaling.md

```diff
@@ -0,0 +1,133 @@
+called tasks.py. First we define our own class derived from the Celery base class:
+
+```python
+class OrchestratorCelery(Celery):
+    def on_init(self) -> None:
+        from orchestrator import OrchestratorCore
+
+        app = OrchestratorCore(base_settings=AppSettings())
+        load_surf(app)  # This will load the workflows
+```
+
+The `load_surf` should be replaced by your own function that at least makes sure that all the workflows are imported
+(which make sure that the are registered) so that the worker can recognize them. This is the minimum implementation
+you need, but you might want to add other initialization that is needed to execute workflows.
+
+Next we instantiate celery using our own `OrchestratorCelery` class:
+
+```python
+broker = f"redis://{AppSettings().CACHE_HOST}:{AppSettings().CACHE_PORT}"
+backend = f"rpc://{AppSettings().CACHE_HOST}:{AppSettings().CACHE_PORT}/0"
+
+celery = OrchestratorCelery("proj", broker=broker, backend=backend, include=["orchestrator.services.tasks"])
+
+celery.conf.update(result_expires=3600)
+```
+
+As you can see in the code above we are using Redis as broker. You can of course replace this by RabbitMQ or 
+another broker of your choice. See the Celery documentation for more details.
+
+`"orchestrator.services.tasks" ` is the namespace in orchestartor-core where the Celery tasks can be found. At the
+moment 4 tasks are defined: 
+
+1. `tasks.new_task`: start a new task (delivered on the Task queue)
+2. `tasks.new_workflow`: start a new workflow (delivered on the Workflow queue)
+3. `tasks.resume_task`: resume an existing task (delivered on the Task queue)
+4. `tasks.resume_workflow`: resume an existing workflow (delivered on the Workflow queue)
+
+
+Finally, we initialize the orchestrator core:
+
+```python
+def init_celery() -> None:
+    from orchestrator.services.tasks import initialise_celery
+
+    initialise_celery(celery)
+
+
+# Needed if we load this as a Celery worker because in that case there is no 'main app'
+init_celery()
+```
+
+The code above sets our local celery instance (which initializes the workflows) as the celery instance that is
+going to be used by the orchestrator-core. Without this code the orchestrator-core would be only aware of a limited
+set of workflows that are part of orchestrator-core itself.
+
+### Running locally
+
+If you want to test your application locally you have to start both the orchestrator-api and one ore more workers.
+For example:
+
+Start the orchestrator api:
+
+```bash
+EXECUTOR="celery" DATABASE_URI="postgresql://nwa:nwa@localhost/nwa-workflows" ./start.sh dev
+```
+
+Notice that we are setting `EXECUTOR` to `celery`. Without that variable the api resorts to the default threadpool.
+
+Start a single worker that listens both on the `tasks` and `workflows` queue:
+
+```bash
+DATABASE_URI="postgresql://nwa:nwa@localhost/nwa-workflows" celery -A surf.tasks  worker --loglevel=info -Q tasks,workflows
```

Comment:
```suggestion
celery -A surf.tasks  worker --loglevel=info -Q tasks,workflows
```

The queues that the orchestrator-api  uses to submit tasks is not configurable, but we need to specify the correct queue names here. Maybe add some explanation what `-Q` is for and that you can use it to make high/low-prio workers, by specifying different queues

Reply:
I'll elaborate a bit on this